### PR TITLE
chore: migrate from lerna to modern tools

### PR DESCRIPTION
**Pull request by mh-alahdadian**
upgrade to yarn 3
migrate from lerna to yarn + nx + changeset

### CODE REVIEW

File: packages/react/package.json [hunk 1]
```diff
@@ -11,17 +11,17 @@
...
     "test": "utsdx test",
-    "lint": "utsdx lint",
-    "prepublishOnly": "yarn lint && yarn test && yarn build",
     "type:check": "yarn tsc"
   },
   "name": "@sahab/react",
   "module": "dist/react.esm.js",
   "devDependencies": {
-    "@testing-library/react-hooks": "^7.0.2",
     "@sahab/utils": "^0.2.0",
```
Comment (mdeljavan) [inline packages/react/package.json]: I know you don''t touch this but move this package to dependencies.

File: lint-staged.config.js [hunk 2]
```diff
@@ -1,11 +1,10 @@
 module.exports = {
```
Comment (mdeljavan) [inline lint-staged.config.js]: IMO use `.json` config file instead of `.js`

File: packages/react/package.json (hunk 1 above)
Comment (mh-alahdadian) [reply packages/react/package.json]: why dependencies ?  It is for testing which is a devDependency

File: lint-staged.config.js (hunk 2 above)
Comment (mh-alahdadian) [reply lint-staged.config.js]: because of yarn type:check command we need to use function model here

**Comment by mh-alahdadian** [comment]
this change will be split into two separate PR 
#10 will handle prettier and lint system improvements
#6 this change will be continued after we discussed about lerna

**Review by mdeljavan — APPROVED** [review]

**Pull request merged**